sympy/polys/matrices/domainmatrix.py

```diff
@@ -1748,4 +1746,5 @@ def lll(A, delta=QQ(3, 4)) -> 'DomainMatrix':
         .. [3] Murray R. Bremner, "Lattice Basis Reduction: An Introduction to the LLL Algorithm and Its Applications"
 
         """
-        return DomainMatrix.from_rep(A.rep.lll(delta=delta))
+        reduced, transform = A.rep.lll(delta=delta, return_transform=return_transform)
+        return DomainMatrix.from_rep(reduced), DomainMatrix.from_rep(transform) if return_transform else None
```

Comment:
This line is too long. Using ternary if else like this is generally less clear than an explicit if block:
```python
        reduced, transform = A.rep.lll(delta=delta, return_transform=return_transform)
        reduced = DomainMatrix.from_rep(reduced)
        if transform is not None:
            transform = DomainMatrix.from_rep(transform)
        return reduced, transform
```
I would generally prefer to have a separate method though like `lll_transform` rather than have a keyword argument that changes the return type.

Reply:
Ok I will change the ternary. I also 100% agree with you that a separate method would be nicer, but then if someone wants the reduced basis and the transform matrix, they have to run the full algorithm twice. As it is now, I have the return type as a tuple of matrices, with the second one being None or the transform matrix. I didn't see a better way to do it all in one go, yet also give the user the option to not calculate the transform matrix at all. What are your thoughts on this? I am rather unopinionated because I think every option has downsides, but maybe you see a way to do it cleanly.

Reply:
I would expect `lll_transform` to return both the reduced basis and the transformation matrix. Basically it looks like this:
```python
def _ddm_lll(a, transform):
    # All the code you already have in ddm_lll
    ...
    if transform:
        return B, T
    else:
        return B

def ddm_lll(a):
    return _ddm_lll(a, transform=False)

def ddm_lll_transform(a):
    return _ddm_lll(a, transform=True)
```
Then DDM, SDM and DomainMatrix can all have `lll` and `lll_transform` methods that ultimately call either `ddm_lll` or `ddm_lll_transform`.

Reply:
Ok I tried to do it this way, I see what you are saying. Please let me know if it looks good now, thanks.